sirepo/sim_data/ml.py

```diff
@@ -104,30 +103,27 @@ def __init__(self, file_path):
     def _get_archive_extension(self):
         x = list(
             filter(
-                lambda s: self._is_archive_type(s),
+                lambda s: self.path.basename.endswith(s),
                 DataReader._SUPPORTED_ARCHIVE_EXTENSIONS,
             )
         )
         return x[0] if len(x) else None
```

Comment:
An empty list in python is False so you can omit the len().

Reply:
[3b9b562](https://github.com/radiasoft/sirepo/pull/4736/commits/3b9b562469fab32bc1a6dbd85b7b1c715dcb140a)